src/bin/download_sysext.rs

```diff
@@ -342,6 +342,57 @@ async fn main() -> Result<(), Box<dyn Error>> {
     fs::create_dir_all(&unverified_dir)?;
     fs::create_dir_all(&temp_dir)?;
 
+    // The default policy of reqwest Client supports max 10 attempts on HTTP redirect.
+    let client = Client::builder().redirect(Policy::default()).build()?;
+
+    // If input_xml exists, simply read it.
+    // If not, try to read from payload_url.
+    let res_local = match args.input_xml {
+        Some(name) => {
+            if name == "-" {
+                Some(io::read_to_string(io::stdin())?)
+            } else {
+                let file = File::open(name)?;
+                Some(io::read_to_string(file)?)
+            }
+        }
+        None => None,
+    };
+
+    let response_text = match (res_local, args.payload_url) {
+        (Some(_), Some(_)) => {
+            return Err("Only one of the options can be given, --input-xml or --payload-url.".into());
+        }
+        (Some(res), None) => res,
+        (None, Some(url)) => {
+            let u = Url::parse(&url)?;
+            let fname = u.path_segments().ok_or(anyhow!("failed to get path segments, url ({:?})", u))?.next_back().ok_or(anyhow!("failed to get path segments, url ({:?})", u))?;
+            let mut pkg_fake: Package;
+
+            let temp_payload_path = unverified_dir.join(fname);
+            pkg_fake = fetch_url_to_file(
+                &temp_payload_path,
+                Url::from_str(url.as_str()).context(anyhow!("failed to convert into url ({:?})", url))?,
+                &client,
+            )
+            .await?;
+            do_download_verify(
+                &mut pkg_fake,
+                output_dir,
+                unverified_dir.as_path(),
+                args.pubkey_file.as_str(),
+                &client,
+            )
+            .await?;
+
+            // verify only a fake package, early exit and skip the rest.
+            return Ok(());
+        }
+        (None, None) => return Err("Either --input-xml or --payload-url must be given.".into()),
+    };
+
+    debug!("response_text: {:?}", response_text);
```

Comment:
The above `let` is not needed and complicates the flow. If you really want to print this, you can do so by printing `res_local` earlier or you can move the print to a match case like `(Some(res), None) => debug!("response text {:?}", res);,`.

Reply:
I do not need a debug message at all. That's not what I care.

But if I do not have `let response_text = ` above, the `parse response` part much below, i.e. `let resp = omaha::Response::from_str(&response_text)?;` cannot be compiled. That's why I cannot drop that.

Reply:
Ok, that part was not shown in the GitHub PR context.
You could still move it here, though (with `match (&res_local), …` instead of `let response_text = match (res_local, …`):
```suggestion
    let response_text = res_local.ok_or(E)?;
    debug!("response_text: {:?}", response_text);
```

Reply:
Done